app/components/widgets/settings/radio_setting.js

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {Text, TouchableOpacity, View} from 'react-native';
+import CheckMark from 'app/components/checkmark';
+import {paddingHorizontal as padding} from 'app/components/safe_area_view/iphone_x_spacing';
+import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
+
+export default class RadioSetting extends PureComponent {
+    static propTypes = {
+        id: PropTypes.string.isRequired,
+        label: PropTypes.node.isRequired,
+        options: PropTypes.array.isRequired,
+        default: PropTypes.string,
+        onChange: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        isLandscape: PropTypes.bool.isRequired,
+        helpText: PropTypes.node,
+        errorText: PropTypes.node,
+    };
+
+    static defaultProps = {
+        isLandscape: false,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value: typeof props.default === 'undefined' ? props.options[0] : props.default,
```

Comment:
I find this confusing.  How about remove `state` here and let the parent component (`DialogElement`) handle it and just pass the selected value to `RadioSetting`?